broker/src/main/java/io/zeebe/broker/system/partitions/impl/StateControllerImpl.java

```diff
@@ -152,6 +162,24 @@ public int getValidSnapshotsCount() {
     return constructableSnapshotStore.getLatestSnapshot().isPresent() ? 1 : 0;
   }
 
+  private void tryCompactState() throws Exception {
+    final var filesBeforeCompaction = runtimeDirectory.toFile().listFiles().length;
```

Comment:
`listFiles()` reports it could be null - could this affect us at all? From the doc:

>  Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.

I guess we ensure it exists and is a directory before, so it's probably fine? :thinking:

Reply:
Yes. That's what I thought. Should we check for null to make Sonarlint happy?

Reply:
It's OK to suppress the warning with a comment, i.e. we already guarantee that the null conditions will not happen.